syntax = "proto3";

import "google/protobuf/empty.proto";

package coldmod_msg.proto.tracing;

message Trace {
  string key = 1;
  string thread_id = 2;
  string process_id = 3;
}

message TraceSrc {
  string key = 1;
}

message TraceSrcs {
  repeated TraceSrc trace_srcs = 2;
}

message ModCommandsArgs {
  string key = 1;
}

message ModCommand {
    string key = 1;
}

service Traces {
  rpc collect(stream Trace) returns (google.protobuf.Empty);
  rpc set(TraceSrcs) returns (google.protobuf.Empty);
  rpc mod_commands(ModCommandsArgs) returns (stream ModCommand);
}
